qemu/tests/qemu_guest_agent.py

```diff
@@ -818,6 +818,31 @@ def _action_after_fsfreeze(self, *args):
             raise error.TestFail("FS freeze failed, guest still can"
                                  " write file")
 
+    def gagent_check_reset_reboot_shutdown(self, test, params, env):
+
+        """
+        Execute "system_reset" and send "shutdown,reboot"
+        command to guest agent after FS freezed
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        vm = env.get_vm(params["main_vm"])
+        vm.verify_alive()
+        gagent = self.gagent
+        gagent.fsfreeze()
+
+        if gagent.shutdown(gagent.SHUTDOWN_MODE_POWERDOWN):
+            test.fail("shutdown command shouldn't succeed for freeze FS")
+        gagent.verify_fsfreeze_status(gagent.FSFREEZE_STATUS_FROZEN)
+
+        if gagent.shutdown(gagent.SHUTDOWN_MODE_REBOOT):
+            test.fail("shutdown command shouldn't succeed for freeze FS")
```

Comment:
should be "reboot command xxx"?

Reply:
and same issue with above, need to check shutdown failure reason

Reply:
1. SHUTDOWN_MODE_REBOOT is equivalent to reboot command.  There is 3 MODEs 1) powerdown 2)reboot 3) halt.

2. shutdown failure reason is that Guest status is frozen.  If guest's status isn't frozen before executing shutdown command(verify_fsfreeze_status(gagent.FSFREEZE_STATUS_FROZEN)), it will raise  VAgentFreezeStatusError  and testing will be aborted.

Reply:
In addition,  function fsfreeze includes verify_fsfreeze_status() function.

Reply:
I mean the test.faill msg better change to "reboot" rather than shutdown, it will maker users more clear about the operation.

Reply:
And for the failure reason, I prefer to get the failure msg from https://github.com/avocado-framework/avocado-vt/pull/1391 and then check if the error include "The command xxx has been disabled for this instance".

Reply:
@lijinlijin 
1. I will replace shutdown with reboot.

2. about failure reason, If so, It is equivalent to hard code.  If qemu monitor return a little different error msg(e.g additional word..etc) in later version. This case will fail.   In fact, I hope this case pass.

Reply:
Could we add it to .cfg file?
Once it changes, modify cfg file is ok.

Reply:
@lijinlijin  I try to match keywords of error msg. e.g
+        except gagent.VAgentCmdError as detail:
+            re.rearch('guest-shutdown has been disabled', detail)

 (1/2) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.4.x86_64.myprovider.qemu_guest_agent.virtio_serial.check_reset_reboot_shutdown_fsfreeze: PASS (91.28 s)
 (2/2) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.4.x86_64.myprovider.qemu_guest_agent.isa_serial.check_reset_reboot_shutdown_fsfreeze: PASS (70.38 s)